plugins/googleanalytics/src/org/pentaho/di/trans/steps/googleanalytics/GaInputStep.java

```diff
@@ -0,0 +1,266 @@
+import org.pentaho.di.trans.step.StepMetaInterface;
+
+
+public class GaInputStep extends BaseStep implements StepInterface {
+
+  private static Class<?> PKG = GaInputStepMeta.class; // for i18n purposes
+
+  private GaInputStepData data;
+  private GaInputStepMeta meta;
+
+  private Analytics analytics;
+  private String accountName;
+
+  public GaInputStep( StepMeta s, StepDataInterface stepDataInterface, int c, TransMeta t, Trans dis ) {
+    super( s, stepDataInterface, c, t, dis );
+  }
+
+  public boolean processRow( StepMetaInterface smi, StepDataInterface sdi ) throws KettleException {
+
+    meta = (GaInputStepMeta) smi;
+    data = (GaInputStepData) sdi;
+
+    if ( first ) {
+
+      first = false;
+
+      data.outputRowMeta = new RowMeta();
+      meta.getFields( data.outputRowMeta, getStepname(), null, null, this, repository, metaStore );
+
+      // stores the indices where to look for the key fields in the input rows
+      data.conversionMeta = new ValueMetaInterface[meta.getFeedField().length];
+
+      for ( int i = 0; i < meta.getFeedField().length; i++ ) {
+
+        // get output and from-string conversion format for each field
+        ValueMetaInterface returnMeta = data.outputRowMeta.getValueMeta( i );
+
+        ValueMetaInterface conversionMeta;
+
+        conversionMeta = ValueMetaFactory.cloneValueMeta( returnMeta, ValueMetaInterface.TYPE_STRING );
+        conversionMeta.setConversionMask( meta.getConversionMask()[i] );
+        conversionMeta.setDecimalSymbol( "." ); // google analytics is en-US
+        conversionMeta.setGroupingSymbol( null ); // google analytics uses no grouping symbol
+
+        data.conversionMeta[i] = conversionMeta;
+      }
+    }
+
+    // generate output row, make it correct size
+    Object[] outputRow = RowDataUtil.allocateRowData( data.outputRowMeta.size() );
+
+    List<String> entry = null;
+
+    // no query prepared yet?
+    if ( data.query == null ) {
+
+      data.query = getQuery( analytics );
+      // use default max results for now
+      //data.query.setMaxResults( 10000 );
+
+      try {
+        data.feed = data.query.execute();
+        data.entryIndex = 0;
+
+      } catch ( IOException e2 ) {
+        throw new KettleException( e2 );
+      }
+
+    } else if ( data.entryIndex >= data.feed.getItemsPerPage() ) {
+      try {
+        // query is there, check whether we hit the last entry and re-query as necessary
+        if ( data.query.getStartIndex() + data.entryIndex <= data.feed.getTotalResults() ) {
+          // need to query for next page
+          data.query.setStartIndex( data.query.getStartIndex() + data.entryIndex );
+
+          data.feed = data.query.execute();
+          data.entryIndex = 0;
+        }
+      } catch ( IOException e2 ) {
+        throw new KettleException( e2 );
+      }
+    }
+
+    List<List<String>> entries = data.feed.getRows();
+    if ( data.entryIndex < entries.size() ) {
+      entry = entries.get( data.entryIndex++ );
+    }
```

Comment:
Shouldn't there be an : 
else {
  entry = null;
}

Old getNextEntry method returned null indicating no more entries...